blog/nse_portfolio_sorts/index.qmd

```diff
@@ -0,0 +1,931 @@
+    siccd >= 9000 & siccd <= 9999 ~ "Public",
+    TRUE ~ "Missing"
+  ))
+
+# Drop NAs
+crsp_monthly <- crsp_monthly |>
+  drop_na(ret_excess, mktcap, mktcap_lag)
+
+# Save 
+crsp_monthly |>
+  dbWriteTable(conn = data_tidy_nse, 
+               name = "crsp_monthly", 
+               value = _,
+               overwrite = TRUE)
+
+
+# Compustat Annual --------------------------------------------------------
+# Load
+compustat <- tbl(wrds, in_schema("comp", "funda")) |>
+  filter(
+    indfmt == "INDL" &
+      datafmt == "STD" &
+      consol == "C" &
+      datadate >= start_date & datadate <= end_date
+  ) |>
+  select(
+    gvkey, # Firm identifier
+    datadate, # Date of the accounting data
+    at, # Total assets
+    ceq, # Total common/ordinary equity
+    ib, # income before extraordinary items 
+    itcb, # Investment tax credit
+    lt, # Total liabilities
+    pstkrv, # Preferred stock redemption value
+    pstkl, # Preferred stock liquidating value
+    pstk, # Preferred stock par value
+    prcc_f, # price close
+    seq, # Stockholders' equity
+    txditc,  # Deferred taxes and investment tax credit
+    txdb # Deferred taxes
+  ) |>
+  collect()
+
+# Manipulate
+compustat <- compustat |>
+  mutate(year = year(datadate),
+         month = floor_date(datadate, "month")) |>
+  group_by(gvkey, year) |>
+  filter(datadate == max(datadate)) |>
+  ungroup()
+
+# Replace negative values of sales, total assets, capital expenditure and inventory with zero
+compustat <- compustat |> 
+  mutate(at = if_else(at < 0, NA_real_, at))
+
+# Save
+compustat |>
+  dbWriteTable(conn = data_tidy_nse, 
+               name = "compustat", 
+               value = _,
+               overwrite = TRUE)
+
+
+# CRSP Compustat link -----------------------------------------------------
+# Load linking table
+ccmxpf_linktable <- tbl(wrds, in_schema("crsp", "ccmxpf_linktable")) |>
+  collect()
+
+# Manipulate
+ccmxpf_linktable <- ccmxpf_linktable |>
+  filter(linktype %in% c("LU", "LC") &
+           linkprim %in% c("P", "C") &
+           usedflag == 1) |>
+  select(permno = lpermno, gvkey, linkdt, linkenddt) |>
+  mutate(linkenddt = replace_na(linkenddt, Sys.Date()))
+
+ccm_links <- crsp_monthly |>
+  inner_join(ccmxpf_linktable, by = "permno", multiple = "all") |>
+  filter(!is.na(gvkey) & (month >= linkdt & month <= linkenddt)) |>
+  select(permno, gvkey, month)
+
+# Add to crsp_monthly
+crsp_monthly <- crsp_monthly |>
+  left_join(ccm_links, by = c("permno", "month"))
+
+# Save 
+crsp_monthly |>
+  dbWriteTable(conn = data_tidy_nse, 
+               name = "crsp_monthly", 
+               value = _,
+               overwrite = TRUE)
+
+
+# Finish chunk ------------------------------------------------------------
+# Remove everything but connection
+rm(list = ls()[which(ls() != "data_tidy_nse")])
+```
+
+
+```{r}
+crsp_monthly <- dbReadTable(data_tidy_nse, "crsp_monthly")
+compustat <- dbReadTable(data_tidy_nse, "compustat")
```

Comment:
Wir nutzen eigentlich immer dbplyr::collect() statt DBI::dbReadTable()

Reply:
Changed accordingly in commit https://github.com/tidy-finance/website/pull/34/commits/7c7a07548ff04a43ecae67d7c3d78843580ff14c